fastplotlib/widgets/image.py

```diff
@@ -703,23 +552,22 @@ def _process_indices(
         indexer = [slice(None)] * self.ndim
 
         numerical_dims = list()
+
+        data_ix = None
+        for i in range(len(self.data)):
+            if self.data[i] is array:
+                data_ix = i
+                break
+        if data_ix is None:
+            raise ValueError(f"Given `array` not found in `self.data`")
```

Comment:
I'm wondering why this is here, this exception should never be reached because this is a method that the user never calls :thinking: